cfgov/v1/atomic_elements/organisms.py

```diff
@@ -281,32 +284,53 @@ class Media:
         js = ['bureau-structure.js']
 
 
-class AtomicTableInput(TableInput):
+class RichTextTableInput(WidgetWithScript, forms.HiddenInput):
+    def __init__(self, table_options=None, attrs=None):
+        super(RichTextTableInput, self).__init__(attrs=attrs)
+        self.table_options = table_options
 
     def render(self, name, value, attrs=None):
-        # Calling the grandparents render method and bypassing TableInputs,
-        # in order to control how we render the form.
-        original_field_html = super(TableInput, self).render(
-            name, value, attrs
+        value = self.json_dict_apply(
+            value,
+            partial(expand_db_html, for_editor=True)
         )
 
+        html = super(RichTextTableInput, self).render(name, value, attrs)
         return Markup(render_to_string('wagtailadmin/table_input.html', {
-            'original_field_html': original_field_html,
+            'original_field_html': html,
             'attrs': attrs,
             'value': value,
         }))
 
     def render_js_init(self, id_, name, value):
-        return "initAtomicTable({0}, {1});".format(
+        return "initRichTextTable({0}, {1});".format(
             json.dumps(id_),
             json.dumps(self.table_options)
         )
 
+    def value_from_datadict(self, data, files, name):
+        value = super(RichTextTableInput, self).value_from_datadict(
+            data, files, name
+        )
+
+        return self.json_dict_apply(value, DbWhitelister.clean)
+
+    @staticmethod
+    def json_dict_apply(value, callback):
+        value = json.loads(value)
+
+        for row in (value or {}).get('data') or []:
```

Comment:
Couldn't this be `for row in (value or {}).get('data', []):`? I had to read the multiple `or`s twice.